content/configuration/artifacts.md

```diff
@@ -0,0 +1,183 @@
+
+### Docker File for First Build Step
+
+The following is a minimal example Docker file to demonstrate the interaction between the artifact and Docker in the first build step:
+
+> **Buildfile**
+```
+FROM nginx
+
+RUN mkdir -p /app/built-files && touch /app/built-files/test.txt && echo "test1234" > /app/built-files/test.txt
+```
+
+This simply creates a directory that matches the path specified in the write artifact i.e "/app/built-files", and then creates a file within the directory to populate the artifact.
+
+{{< note title="Note" >}}
+In practice the contents of the "/app/built-files" directory would be the result of a build process.
+{{< /note >}}
+
+### Docker File for Second Build Step
+
+The following is a minimal example Docker file to demonstrate the interaction between the artifact and Docker in the second build step:
+
+> **Dockerfile**
+```
+FROM nginx
+
+COPY . /usr/share/nginx/html
+COPY ./built-files /usr/share/nginx/html
+```
+
+This uses the Docker [COPY instruction](https://docs.docker.com/engine/reference/builder/#copy) to import files into the image, copying them to the web server directory. The first COPY instruction imports code from the project repository as normal. The second COPY instruction imports any files shared in the artifact by referencing the path specified in the read artifact prefixed by a full stop i.e "./built-files". In this case there is just one file in the artifact, which will be copied from "./built-files/test.txt" to "/usr/share/nginx/html/test.txt".
+
+## Using Artifacts With Secrets
+
+If your initial build image needs to use secure access credentials (such as an auth token) you can supply it as an environment variable:
+
+``` yaml
+tasks:
+    images:
+        build:
+            services:
+                web:
+                    steps:
+                        - docker_file_path: ./Buildfile
+                          environment:
+                              - name: GITHUB_TOKEN
+                                value: ${GITHUB_TOKEN}
+                          write_artifacts:
+                              - name: built-files-artifact
+                                path: /app/built-files
+                        - docker_file_path: ./Dockerfile
+                          image: ${IMAGE_NAME}
+                          read_artifacts:
+                              - name: built-files-artifact
+                                path: /built-files
+```
+
+Here the image in the first build step will be supplied with the enviroment variable `GITHUB_TOKEN`, but the image in the second build step will have no record of it.
+
+The following is an example Docker file to demonstrate how the `GITHUB_TOKEN` will be used in the first build step:
+
+> **Buildfile**
+```
+FROM nginx
```

Comment:
I'd use a PHP based image to have GITHUB_TOKEN, I'd say. `quay.io/continuouspipe/symfony-php7.1-nginx:stable` as base?